Improvements All Around (READ DESCRIPTION)

CDAGaming opened the pull request:
This PR does the following:
- Updates Forge, Mappings, and JEI to their Latest Version
  - This includes Naming Changes and some other nifty details too.
- Upgrades Gradle to 4.3-bin (WIP)

CDAGaming committed:
Improvements All Around

This PR does the following:
- Updates Forge, Mappings, and JEI to their Latest Version
  - This includes Naming Changes and an additional fix to allow development versions of Forge to be ran with this mod (Might re-add the latest version back to required just in case specific 1.12.2 versions of forge break this mod or this PR)
- Switches Gradle to 4.0-all for Usage with IDE's like IntelliJ
 - I held off on Updating to a Higher Gradle Version, just in case.

Tamaized commented on src/main/java/twilightforest/TwilightForestMod.java:
```diff
@@ -28,7 +28,7 @@
 		version = TwilightForestMod.VERSION,
 		acceptedMinecraftVersions = "[1.12.2]",
-		dependencies = "after:ctm@[MC1.12-0.2.3.9,);required-after:forge@[14.23.0.2501,)")
+		dependencies = "after:ctm@[MC1.12-0.2.3.9,);required-after:forge@[14.23.0.0,)")
```
?

Drullkus commented on src/main/java/twilightforest/TwilightForestMod.java:
```diff
@@ -28,7 +28,7 @@
 		version = TwilightForestMod.VERSION,
 		acceptedMinecraftVersions = "[1.12.2]",
-		dependencies = "after:ctm@[MC1.12-0.2.3.9,);required-after:forge@[14.23.0.2501,)")
+		dependencies = "after:ctm@[MC1.12-0.2.3.9,);required-after:forge@[14.23.0.0,)")
```
Absolutely not

CDAGaming replied:
This change was to allow development versions of Forge to be used with this mod/any 1.12.2 Forge Version.

For this, what do you suggest I do here, my idea was if this isn't allowed, to instead use the Forge Version it's build with here.

Drullkus replied:
Forge build `14.23.0.2496` introduces the ability to add custom map icons, we'd prefer to forge the forge version to any version after this or else Magic Maps will render with bad structure icons.

CDAGaming replied:
Ah, Ok. I see. In that case, ill set the required build to any one after 14.23.0.2496 in the next commit to this PR (Next Commit will update MCP again to Today's Release from Yesterday's)

Meanwhile, are there any other concerns on this PR? (Mainly with the MCP Changes and other dependencies)

Drullkus replied:
Right now my next concern is that I will be forced to re-run setup as well as 4 other people, so I'll hold until CTM gets an update since I actually do need a thing in the coming CTM version.

CDAGaming replied:
*facepalm* Completly forgot to change this line lol

After this PR, it'll now require Forge 14.23.0.2496 or higher 👍

CDAGaming replied:
*Proceeds to mess up further by still not changing the line* Will fix this today with MCP Follow up to keep this PR maintained.

CDAGaming commented:
Have been on Vacation the last week, continuing this PR and will resolve conflicts

CDAGaming committed:
Merge branch '1.12.x' into 1.12.x

CDAGaming commented:
Conflicts Resolved, will Update Mappings on this PR hopefully later today

CDAGaming committed:
Update MCP Mapping to 20171021

CDAGaming committed:
Forgot to Update Forge Also...

CDAGaming committed:
Forgot JEI's Update Also *facepalm*

CDAGaming committed:
Merge pull request #1 from TeamTwilight/1.12.x

Upstream from origin/1.12.x

CDAGaming commented on gradle/wrapper/gradle-wrapper.properties:
```diff
@@ -1,6 +1,6 @@
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-4.0-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-4.0-all.zip
```
Since this is on 4.0, i might just update this to 4.2 as it should be more stable (Will test this to be sure it still properly builds)

CDAGaming replied:
4.0-all to 4.2.1-all to be precise (Unless 4.3 releases within the next day xD)

Drullkus replied:
Why switch Gradle from bin to all?

CDAGaming replied:
Tbh, idk myself. Seems IntelliJ IDEA always suggest it's better to use -All, but since I don't think it's really needed I'll probably rebert that

williewillus commented:
idk. in general I don't see the point of bumping mappings unless
1. Porting to next MC version (or preparing to do so)
2. There is a critical new subsystem (loot tables, advancements, etc.) that wasn't mapped yet but now is.

Except that 2 almost always happens alongside 1.
As of right now it's just churn, which will have to happen anyway when 1.13 lands.

Also, what was the gradle upgrade mentioned? I see 4.0 in the main repo

Tamaized commented:
I feel the same here, there really isn't a real point to the PR as of right now, Drull plans to update stuff when the new CTM version arrives anyway so may as well just hold off till then.

Drullkus commented:
Agreed with Willie and Tam

CDAGaming committed:
Merge pull request #2 from TeamTwilight/1.12.x

Update to Mod v3.3 (AWAIT MCP FOLLOWUP COMMIT)

CDAGaming commented:
Am going to Update this PR to the latest update of forge later today (22 PRs merged on their repository in less than an hour, so ill need to recheck all my PRs and projects  with forge to make sure stuff works still)

Tamaized commented:
Is there any actual good reasoning for this PR. What good does it do for the codebase? Forge version and mapping updates won't change anything. Updating those should be left to the maintainers anyway.

CDAGaming commented:
Well at this time not very much besides a few renames in recent mapping updates, but yea i think your right.

CDAGaming commented:
To-Do:

- [x] Update Forge, MCP, and Gradle (Gradle going to 4.3-bin)
- [x] Revert Allowing Dev Versions of Forge
- [x] Push PR to CTM for Official/Better Compatibility with this PR and 1.12.2 in general
- [ ] Optional Code Cleanup (Wait until Tomorrow)

CDAGaming committed:
Merge pull request #3 from TeamTwilight/1.12.x

Upstream/origin

CDAGaming committed:
Updates to Dependencies + Fixes

Updated:

- Forge, MCP, JEI
- Gradle Wrapper to 4.3-bin

Fixes:
- Fix Allowing Development Versions of Forge to be run
(Can break the Mod if under 2496)

CDAGaming committed:
Thx Git for not Pushing this with Prior Commit :/

CDAGaming commented:
This PR is officially Clear to Merge, unless you'd like the Code Cleanup stuff to be combined in this PR as well
